test/elixir_bench_web/views/job_view_test.exs

```diff
@@ -0,0 +1,52 @@
+defmodule ElixirBenchWeb.JobViewTest do
+  use ElixirBenchWeb.ConnCase, async: true
+
+  # Bring render/3 and render_to_string/3 for testing custom views
+  import Phoenix.View
+
+  test "render index.json given job without docker dependencies and environment variables" do
+    repo = build(:repo, %{owner: "owner", name: "name"})
+    job = build(:job, repo: repo)
+
+    %{uuid: uuid, branch_name: branch, commit_sha: commit_sha, config: config} = job
+    %{elixir: elixir, erlang: erlang} = config
+
+    %{data: [job_data]} = render(ElixirBenchWeb.JobView, "index.json", %{jobs: [job], repo: repo})
+
+    assert %{
+             branch_name: ^branch,
+             commit_sha: ^commit_sha,
+             config: %{
+               deps: %{docker: []},
+               elixir_version: ^elixir,
+               environment_variables: %{},
+               erlang_version: ^erlang
+             },
+             id: ^uuid,
+             repo_slug: "owner/name"
+           } = job_data
```

Comment:
isn't this already tested above in the controller test? I think if we already test the view in the controller test (which is fine imo) then we don't need to duplicated that test logic in the view code or am I missing a nuance here?

When I got controller tests I often just use view tests to test edge cases that don't matter for the controller to have them setup more easily

Reply:
I had difficult to decide what should I test in each level. I kinda saw it with this perspective. We have mainly two specific cases that could break the View 
1) when job does not have docker deps
2) when job has docker deps

Then I replicated the test in the controller just to ensure its calling the view and the "integration" is working. Did that because I think that have/not have deps should not be concern of the controller.